espree.js

```diff
@@ -799,107 +800,120 @@ function scanNumericLiteral() {
     };
 }
 
-// 7.8.4 String Literals
+/**
+ * Scan a string escape sequence and return its special character.
+ * @param {string} ch The starting character of the given sequence.
+ * @returns {Object} An object containing the character and a flag
+ * if the escape sequence was an octal.
+ * @private
+ */
+function scanEscapeSequence(ch) {
+    var code,
+        unescaped,
+        restore,
+        escapedCh,
+        octal = false;
+
+    // An escape sequence cannot be empty
+    if (!ch) {
+        throwError({}, Messages.UnexpectedToken, "ILLEGAL");
```

Comment:
The only change this refactoring made to the behavior itself was to throw an error if this case is reached. If you check back in the removed code, it looks like this case currently gets lost in the pile of if/elses and goes unchecked, when it should result in an error. 

Does this sound correct? A second pair of eyes couldn't hurt